src/lib.rs

```diff
@@ -182,9 +182,10 @@ pub trait Manager: Send + Sync + 'static {
     /// succeeds.
     async fn check(&self, conn: Self::Connection) -> Result<Self::Connection, Self::Error>;
 
-    /// Determines connection is still valid when check-in.
-    async fn test_on_check_in(&self, _conn: &mut Self::Connection) -> Result<(), Self::Error> {
-        Ok(())
+    /// *Quickly* determines a connection is still valid when check-in.
+    #[inline]
+    fn is_conn_valid(&self, _conn: &mut Self::Connection) -> bool {
```

Comment:
For symmetry with `check` and `connect` this may be called `validate`?

Reply:
Nice name!